packages/public/default-document-editor/src/index.tsx

```diff
@@ -111,12 +112,14 @@ export function createDefaultDocumentEditor(
                     <EdtrIcon icon={edtrClose} />
                   </BorderlessOverlayButton>
                 </Header>
-                {renderSettings(children, { close })}
+                {renderSettings
+                  ? renderSettings(children, { close })
+                  : children}
               </React.Fragment>
             )
           }
         : undefined
-    }, [renderSettings])
+    }, [renderSettings, showSettings()])
```

Comment:
That won't work

Reply:
Why? I want do update, whenever showSettings() returns a different value.